core/src/main/scala/commandcenter/command/LoremIpsumCommand.scala

```diff
@@ -0,0 +1,73 @@
+package commandcenter.command
+
+import cats.data.Validated
+import cats.syntax.apply._
+import com.monovore.decline
+import com.monovore.decline.Opts
+import commandcenter.CommandContext
+import commandcenter.util.ProcessUtil
+import commandcenter.view.DefaultView
+import io.circe.Decoder
+import scala.io.Source
+import zio._
+
+sealed trait ChunkType
+
+case object Word extends ChunkType
+
+case object Sentence extends ChunkType
+
+case object Paragraph extends ChunkType
+
+final case class LoremIpsumCommand() extends Command[Unit] {
+  val commandType: CommandType = CommandType.LoremIpsumCommand
+
+  val commandNames: List[String] = List("lipsum", "lorem", "ipsum")
+
+  val title: String = "Lorem Ipsum"
+
+  val lipsum = Source.fromResource("lipsum").getLines().mkString("\n")
+
+  val numOpt = Opts.argument[Int]("number").withDefault(1)
+  val typeOpt = Opts.argument[String]("words, sentences, or paragraphs").mapValidated {
+    case arg if arg.matches("words?") => Validated.valid(Word)
+    case arg if arg.matches("sentences?") => Validated.valid(Sentence)
+    case arg if arg.matches("paragraphs?") => Validated.valid(Paragraph)
+    case s => Validated.invalidNel(s"${s} is not valid: should be 'words', 'sentences', or 'paragraphs'.")
+  }.withDefault(Paragraph)
+
+  val lipsumCommand = decline.Command(title, title)((numOpt, typeOpt).tupled)
```

Comment:
I tried to update this to allow commands like `lorem sentence` by changing this line to:
```scala
val lipsumCommand = decline.Command(title, title)((numOpt, typeOpt).tupled.orElse(typeOpt.map((1, _))))
```
That doesn't work unfortunately - typing in `lorem sentence` gives you an error message of "Invalid integer: sentence".

Reply:
Right, those are positional arguments so an Int argument alone wouldn't be able to cover both cases. You would either need to change the argument to a String and validate from there. Or possibly try making it an Either. But I think that might overcomplicate things.